### PR TITLE
Ruby Capstone Project: Catalog of my things

### PR DESCRIPTION
**Pull request by jaimevillegas**
Hi there!

We, @Alejandro-Bernal-M, @johnie261, and @jaimevillegas, have worked hard to complete all the project requirements.

- Add a descriptive README and LICENSE file. 📄
- Add all classes and methods according to the UML diagram. 📝
- Follow best coding practices like KISS, YAGNI, and SOLID principles. ✨
- Implement read and write functionality with JSON data files. 🔀
- Write tests for each class. ✅
- Create `schema.sql` with tables representing all the classes of the project. 📄
- Build an interactive console app that will receive user input and will show information. 👽️
- Follow gitflow best practices. 💥
- Work as a team with a Kanban Project. 👥
- Make pull requests and review partner's code. 🔍️
- Create a video showing the project functionality. 📸

Thank you so much for taking the time to review our code! ⚡️🚀

### CODE REVIEW

**Review by codecaiine — APPROVED** [review]

File: music_album.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  attr_accessor :genre, :author, :source, :label, :archived
+  attr_reader :id, :publish_date, :on_spotify
+
+  def initialize(id, publish_date, on_spotify)
+    super(publish_date, id)
+    @on_spotify = on_spotify
+  end
+
+  def can_be_archived?
+    super && @on_spotify
```
Comment (codecaiine) [inline music_album.rb:13]: - [ ] According to the project requirements, **can_be_archived?()** should return true if the parent's method returns true AND if on_spotify equals true. Please fix it in order to make your app works as expected 👍 

```ruby
   def can_be_archived?
      super && @on_spotify == true
    end
```

File: item.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  def can_be_archived?
+    Date.today.year - Date.parse(@publish_date).year > 10
+  end
+
+  def move_to_archive
+    @archived = true if can_be_archived?
+  end
+
```
Comment (codecaiine) [inline item.rb:20]: - [ ] According to the project requirements there is a relationship between the Item and the author, which means they should have a method that makes a connection and is present in the Item class. I would like to recommend creating a method for that. Please take a look at the example below ⬇️  and make the change in order to make your project works as expected 👍 


```ruby
    def author=(author)
      @author = author
      author.items.push(self) unless author.items.include?(self)
    end
```

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Good relationship setup ✔️ 
- Test are well implemented ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: music_album.rb (hunk 1 above)
Comment (Alejandro-Bernal-M) [reply music_album.rb:13]: hello @codecaiine it is the same, We are using short circuit and is the same as you said

File: item.rb (hunk 2 above)
Comment (Alejandro-Bernal-M) [reply item.rb:20]: Changes implemented

Comment (Alejandro-Bernal-M) [reply item.rb:20]: Changes implemented

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @TEAM     , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉  Keep up the good work 🥂 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] No linter errors
- [x] Console App works as intended💯
- [x] Tests are passing 🥇  
 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**